### PR TITLE
[Windows] Whois support

### PR DESCRIPTION
**Pull request by gpotter2**
Whois on scapy was only using the `whois` command.
As it's not available on windows, I've added a util function to perform the request.

Edit: updated with my own code

### CODE REVIEW

**Comment by p-l-** [comment]
I don't think we can accept code with a different copyright & license, particularly in the main Scapy code.

**Comment by gpotter2** [comment]
Ok i've done my own code

File: scapy/utils.py [hunk 1]
```diff
@@ -1272,3 +1272,29 @@ def make_lined_table(*args, **kargs):
...
 
+###############################################
+### WHOIS CLIENT (not available on windows) ###
+###############################################
+
+def whois(ip_address):
+    """Whois client for python"""
+    query = "n " + ip_address
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.connect(("whois.arin.net", 43))
```
Comment (guedou) [inline scapy/utils.py]: I would prefer to use whois.ripe.net instead as I tends to provide better/more accurate results

File: scapy/utils.py [hunk 2]
```diff
@@ -1272,3 +1272,29 @@ def make_lined_table(*args, **kargs):
 def make_tex_table(*args, **kargs):
     __make_table(lambda l: "%s", lambda l: "& %s", "\\\\", seplinefunc=lambda a,x:"\\hline", *args, **kargs)
 
+###############################################
+### WHOIS CLIENT (not available on windows) ###
+###############################################
+
+def whois(ip_address):
+    """Whois client for python"""
```
Comment (guedou) [inline scapy/utils.py]: s/python/Python/g

File: scapy/utils.py [hunk 3]
```diff
@@ -1272,3 +1272,29 @@ def make_lined_table(*args, **kargs):
...
+        answer += d
+        if not d:
+            break
+    s.close()
+    result = "#\
+# ARIN WHOIS data and services are subject to the Terms of Use\n\
+# available at: https://www.arin.net/whois_tou.html\n\
+\n\
+"
+    for line in answer.split("\n"):
```
Comment (guedou) [inline scapy/utils.py]: could you replace this with a list comprehension ? 
Like,
lines = [ line for line in answer.split("\n") if line and not line.startswith("#")]
return "\n".join(lines)

Comment (gpotter2) [inline scapy/utils.py]: If we use the .ripe.net server, we won't need it anyway

**Comment by gpotter2** [comment]
@guedou That should do it

File: scapy/base_classes.py [hunk 4]
```diff
@@ -76,7 +76,8 @@ def __init__(self, net):
         self.repr=net
         self.parsed,self.netmask = self._parse_net(net)
 
-
+    def __str__(self):
```
Comment (guedou) [inline scapy/base_classes.py]: Why is this needed ?

Comment (gpotter2) [inline scapy/base_classes.py]: To support `IP(src="google.fr").whois()`

**Comment by gpotter2** [comment]
@guedou So, how about this ?

**Comment by guedou** [comment]
It looks ok !

The gethostbyname seems useless as it only provides a shortcut. I am afraid that it won't be used elsewhere. Therefore, I think that it can be removed.

The whois function could use some checks to avoid throwing exceptions on errors. I don't think that it must be improved right now.

**Comment by gpotter2** [comment]
@guedou That should be good to merge
